lib/features/sks-menu/data/repository/sks_menu_repository.dart

```diff
@@ -7,6 +7,30 @@ import "../models/sks_menu_response.dart";
 
 part "sks_menu_repository.g.dart";
 
+@riverpod
```

Comment:
It's nice to see you've learnt some riverpod! It's very nice to see. Hovewer, riverpod is best about handling global/business state and here it's more about local/emphermal state and in our app we try to use Flutter Hooks for local state scenarios. We still have some old part of the code that use rivepod for semi-global state, but even so then it needs to be scoped to the route. And scoping riverpod is kinda nasty. I would reccommend using hooks for this scenario. 

https://riverpod.dev/docs/essentials/do_dont#avoid-using-providers-for-ephemeral-state: 

<img width="560" alt="Zrzut ekranu 2024-12-2 o 14 37 10" src="https://github.com/user-attachments/assets/67a14a0c-61df-4a33-bd47-35dcdbc5f160">



PS. even if we stay with the riverpod, it needs to be scoped to the root and `repository` is not a proper file for this. Since it's UI related state, it belongs to presentation layer. In case of riverpod you could make `something_controller.dart` file - or just use flutter hooks. They should be pretty easy to implement ;)